internal/pkg/checks/check.go

```diff
@@ -14,86 +14,101 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// Ingress conformance test harness machinery
 package checks
 
 import (
 	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"net/http"
 	"time"
 )
 
+// Config contains test suite configuration fields
 type Config struct {
 }
 
+// Check represents a test case. Checks are named, and must provide a
+// description and a Run function. Checks are organized in a hierarchy.
 type Check struct {
 	Name        string
 	Description string
 
-	Run func(check *Check, config Config) (bool, error)
+	Run func(check *Check, config Config) (success bool, err error)
 
-	// All checks
+	// Child checks
 	checks []*Check
-
 	// Parent check
 	parent *Check
 }
 
+// CapturedRequest contains the original HTTP request metadata as received
+// by the echoserver handling the test request.
+// The DownstreamServiceId field contains the TEST_ID environment variable
+// value of the downstream echoserver.
 type CapturedRequest struct {
-	StatusCode int
-	TestId     string
-	Path       string
-	Host       string
+	DownstreamServiceId string `json:"testId"`
```

Comment:
Document the field with a comment doc (instead of above)

// DownstreamServiceID is ...